### PR TITLE
Fixes #15165 - add facet api documentation

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
Note this requires a foreman change which has not been merged or opened yet

**Comment by jlsherrill** [comment]
Note this makes all of this work in hammer:

```
$ hammer -u admin -p changeme  -r host update --lifecycle-environment=Library --content-view="briantest"  --id=2  --organization-id=1
Host updated
```

**Comment by ShimShtein** [comment]
:+1: The documentation is generated correctly in both dev and production environments.
Tested by running:
``` sh
FOREMAN_APIPIE_LANGS=en RAILS_ENV=production rake apipie:cache

FOREMAN_APIPIE_LANGS=en RAILS_ENV=development rake apipie:cache
```

File: app/controllers/katello/api/v2/host_subscriptions_controller.rb [hunk 1]
```diff
@@ -4,6 +4,13 @@ class Api::V2::HostSubscriptionsController < Katello::Api::V2::ApiController
     before_filter :check_subscriptions, :only => [:add_subscriptions, :remove_subscriptions]
     before_filter :find_content_view_environment, :only => :create
 
+    def_param_group :subscription_facet_attributes do
+      param :release_version, String, :desc => N_("Release version for this Host to use (7Server, 7.1, etc)")
+      param :autoheal, :bool, :desc => N_("Sets whether the Host will autoheal subscriptions upon checkin")
+      param :service_level, Integer, :desc => N_("Service level to be used for autoheal.")
+      param :hypervisor_guest_uuids, Array, :desc => N_("List of hypervisor guest uuids")
```
Comment (thomasmckay) [inline app/controllers/katello/api/v2/host_subscriptions_controller.rb]: param :facts, Hash, :desc => N_("Key-value hash of subscription-manager facts, nesting uses a period delimiter (.)")

Comment (thomasmckay) [inline app/controllers/katello/api/v2/host_subscriptions_controller.rb]: param :installed_products, Array, :desc => N_("List of products installed on the host") do
      param :product_id, String, :desc => N_("Product id as listed from a host's installed products, \
        this is not the same product id as the products api returns")
      param :product_name, String, :desc => N_("Product name as listed from a host's installed products")
      param :arch, String, :desc => N_("Product architecture")
      param :version, String, :desc => N_("Product version")
    end

Comment (jlsherrill) [inline app/controllers/katello/api/v2/host_subscriptions_controller.rb]: I don't know that we can include installed_products without exposing those in hammer:

 --product PRODUCT_NAME                              Name to search by
 --product-id PRODUCT_ID                             product numeric identifier

That might have to be something we add later when we have more controller over the host update hammer command.

Comment (thomasmckay) [inline app/controllers/katello/api/v2/host_subscriptions_controller.rb]: I'm alright w/ leaving off the additional apipie params and opening redmine to track.

**Comment by thomasmckay** [comment]
ACK w/ optional apipie additional params

**Comment by jlsherrill** [comment]
Opened issue:  http://projects.theforeman.org/issues/15202

**Pull request merged**